src/sparsezoo/v2/test.md

```diff
@@ -0,0 +1,111 @@
+model = Model(path=stub)
+model.download() # optional: may be called to download all the files, but is not required (see below). Will be omitted in further examples
+recipes: str = model.recipes.downloaded_path() # will automatically download the file (if not already downloaded) and return the path
+...
+```
+```python
+from sparsezoo import Model
+
+stub = "zoo:cv/classification/mobilenet_v1-1.0/pytorch/sparseml/imagenet/pruned-moderate"
+stub = stub + "?" + "checkpoint=preqat"
+# will read all the files from the server, except for the full training folder. Training folder
+# will be identical to prequat checkpoint directory
+model = Model(path=stub)
+training: str = model.training.downloaded_path() 
+...
+```
+### Through the API purely
+```python
+from sparsezoo import Model
+
+stub = "zoo:cv/classification/mobilenet_v1-1.0/pytorch/sparseml/imagenet/pruned-moderate"
+model = Model(path=stub)
+# ALTERNATIVE ONE: filtering on download
+model.recipes.download(recipe="transfer_learn")
+recipe: str = model.recipes.downloaded_path()
+# ALTERNATIVE TWO: filtering on .downloaded_path()
+recipe: str = model.recipes.downloaded_path(recipe="transfer_learn")
+```
+
+```python
+from sparsezoo import Model
+
+stub = "zoo:cv/classification/mobilenet_v1-1.0/pytorch/sparseml/imagenet/pruned-moderate"
+model = Model(path=stub)
+# ALTERNATIVE ONE: filtering on download
+model.training.download(checkpoint="preqat")
+training: str = model.training.downloaded_path()
+# ALTERNATIVE TWO: filtering on .downloaded_path()
+training: str = model.training.downloaded_path(checkpoint="preqat")
+```
+
+## Download multiple chosen files (also in parallel)
+
+### Through the string arguments
+```python
+from sparsezoo import Model
+
+stub = "zoo:cv/classification/mobilenet_v1-1.0/pytorch/sparseml/imagenet/pruned-moderate"
+# choose the files through the string arguments
+stub = stub + "?" + "recipe=transfer_learn" + "&" + "checkpoint=preqat" + "&" + "recipe=original"
+model = Model(path=stub)
+recipes: List[str] = model.recipes.downloaded_path() 
+training: str = model.training.downloaded_path() # training will be a Directory with a preqat checkpoint
+...
+```
+### Through the API purely
+```python
+from sparsezoo import Model
+
+stub = "zoo:cv/classification/mobilenet_v1-1.0/pytorch/sparseml/imagenet/pruned-moderate"
+model = Model(path=stub)
+# explicitly define which files to download
+model.recipes.download()
+model.training.download(checkpoint="preqat")
+
+recipes: List[str] = model.recipes.downloaded_path() # downloaded_path() may be either of type: str or List[str]
+training: str = model.training.downloaded_path() 
+```
+
+## Download all files, no "filtering"
+```python
+from sparsezoo import Model
+
+stub = "zoo:cv/classification/mobilenet_v1-1.0/pytorch/sparseml/imagenet/pruned-moderate"
+model = Model(path=stub)
+model.download()
+training: str = model.training.downloaded_path() 
+deployment: str = model.deployment.downloaded_path()
```

Comment:
if deployment arg (future support) given, should support model.deployment.default.path